qiskit_experiments/curve_analysis/fit_function.py

```diff
@@ -13,11 +13,40 @@
 """
 A library of fit functions.
 """
-# pylint: disable=invalid-name
+# pylint: disable=invalid-name, no-member
+
+import functools
+from typing import Callable
 
 import numpy as np
+from uncertainties import unumpy as unp
+
+
+def calc_uncertainties(fit_func: Callable) -> Callable:
```

Comment:
We can probably find a better name for this. Perhaps `cast_no_uncertainty_to_float` ?

Reply:
What about https://github.com/Qiskit/qiskit-experiments/pull/551/commits/9e9715bb12b11ab208b5d55353486a93568e501e?